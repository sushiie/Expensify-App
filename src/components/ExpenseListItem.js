import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <Link to={`/edit/${id}`} className="list-item">
        <div>
            <h3 className="list-item__title">{description}</h3>
            <span className="list-item__sub-title">
                {numeral(amount/100).format('$0,0.00')}
            </span>
        </div>
        <h3 className="list-item__data">
            {moment(createdAt).format('MMMM Do, YYYY')}
        </h3>
    </Link>
);

export default ExpenseListItem;