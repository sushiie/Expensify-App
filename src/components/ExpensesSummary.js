import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import totalAmount from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expensesTotal}) =>{
        const expensesWord = expensesCount > 1 ? 'expenses' : 'expense';
        const formattedAmount = numeral(expensesTotal/100).format('$0,0.00');
        return (
            <div>
                {   
                    (<h1>Viewing {expensesCount} {expensesWord} totalling {formattedAmount}</h1>)
                }
            </div>
        );
};
const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: totalAmount(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);