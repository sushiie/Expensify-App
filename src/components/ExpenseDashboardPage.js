import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
    <div>
        This is Expense Dashboard Page
        <ExpensesSummary />
        <ExpenseListFilter />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;