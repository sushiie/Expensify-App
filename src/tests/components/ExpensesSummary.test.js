import { shallow } from 'enzyme';
import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should display message for 1 expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1000}/>);
    expect(wrapper).toMatchSnapshot();
});


test('Should display message for multiple expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={5} expensesTotal={8900}/>);
    expect(wrapper).toMatchSnapshot();
});