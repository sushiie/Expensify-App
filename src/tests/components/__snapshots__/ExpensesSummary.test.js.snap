// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display message for 1 expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $10.00
  </h1>
</div>
`;

exports[`Should display message for multiple expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    5
     
    expenses
     totalling 
    $89.00
  </h1>
</div>
`;
